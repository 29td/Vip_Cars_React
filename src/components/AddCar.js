import Navbar from './Navbar';

const AddCar = () => (
  <div className="App">
    <Navbar />
    <main className="main">
      <h1>This is the Add Car page</h1>
    </main>
  </div>
);

export default AddCar;
