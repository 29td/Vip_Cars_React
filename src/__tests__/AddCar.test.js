import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/';
import { BrowserRouter as Router } from 'react-router-dom';
import { AuthProvider } from 'react-auth-kit';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import AddCar from '../components/AddCar';

describe('Login component', () => {
  test('Should render the proper Login component', () => {
    const login = render(
      <React.StrictMode>
        <Provider store={store}>
          <AuthProvider
            authType="cokie"
            authName="_auth"
            cookieDomain={window.location.hostname}
            cookieSecure={false}
          >
            <Router>
              <AddCar />
            </Router>
          </AuthProvider>
        </Provider>
        ,
      </React.StrictMode>,
    );
    expect(login).toMatchSnapshot();
  });
});
