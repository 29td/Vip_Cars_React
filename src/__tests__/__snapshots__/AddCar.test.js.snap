// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Should render the proper Login component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="Navbar"
        >
          <img
            alt="Vip Cars Logo"
            class="logo"
            src="vip_cars_logo.png"
          />
          <div>
            <ul
              class="navbar-ul"
            >
              <li>
                <a
                  class="navbar-li"
                  href="/"
                  id="active"
                >
                  CARS
                </a>
              </li>
              <li>
                <a
                  class="navbar-li"
                  href="/add_car"
                  id=""
                >
                  ADD CAR
                </a>
              </li>
              <li>
                <a
                  class="navbar-li"
                  href="/delete_car"
                  id=""
                >
                  DELETE CAR
                </a>
              </li>
              <li>
                <a
                  class="navbar-li"
                  href="/reservations"
                  id=""
                >
                  MY RESERVATIONS
                </a>
              </li>
              <li>
                <a
                  class="navbar-li"
                  href="/reserve"
                  id=""
                >
                  RESERVE
                </a>
              </li>
            </ul>
          </div>
          <button
            class="sign-out-btn"
            type="button"
          >
            Log Out
          </button>
          <div
            class="footer"
          >
            <div
              class="NavFooter"
            >
              <footer
                class="text-center text-white"
              >
                <div
                  class="container pt-4"
                >
                  <section
                    class="mb-4"
                  >
                    <a
                      class="btn btn-link btn-floating btn-lg text-dark"
                      data-mdb-ripple-color="dark"
                      href="#!"
                      role="button"
                    >
                      <i
                        class="fa fa-twitter"
                      />
                    </a>
                    <a
                      class="btn btn-link btn-floating btn-lg text-dark"
                      data-mdb-ripple-color="dark"
                      href="#!"
                      role="button"
                    >
                      <i
                        class="fa fa-facebook-f"
                      />
                    </a>
                    <a
                      class="btn btn-link btn-floating btn-lg text-dark"
                      data-mdb-ripple-color="dark"
                      href="#!"
                      role="button"
                    >
                      <i
                        class="fa fa-google-plus"
                      />
                    </a>
                    <a
                      class="btn btn-link btn-floating btn-lg text-dark"
                      data-mdb-ripple-color="dark"
                      href="#!"
                      role="button"
                    >
                      <i
                        class="fa fa-vimeo"
                      />
                    </a>
                    <a
                      class="btn btn-link btn-floating btn-lg text-dark"
                      data-mdb-ripple-color="dark"
                      href="#!"
                      role="button"
                    >
                      <i
                        class="fa fa-pinterest"
                      />
                    </a>
                    <a
                      class="btn btn-link btn-floating btn-lg text-dark"
                      data-mdb-ripple-color="dark"
                      href="#!"
                      role="button"
                    >
                      <i
                        class="fa fa-github"
                      />
                    </a>
                  </section>
                </div>
              </footer>
            </div>
          </div>
        </header>
        <main
          class="main"
        >
          <div
            class="session"
          >
            <section
              class="not-allowed"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-triangle-exclamation caution"
                data-icon="triangle-exclamation"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7 .2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8 .2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24V296c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32z"
                  fill="currentColor"
                />
              </svg>
              <h2>
                Only Admins Can Add A Car!!!!!
              </h2>
            </section>
          </div>
        </main>
      </div>
      ,
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="Navbar"
      >
        <img
          alt="Vip Cars Logo"
          class="logo"
          src="vip_cars_logo.png"
        />
        <div>
          <ul
            class="navbar-ul"
          >
            <li>
              <a
                class="navbar-li"
                href="/"
                id="active"
              >
                CARS
              </a>
            </li>
            <li>
              <a
                class="navbar-li"
                href="/add_car"
                id=""
              >
                ADD CAR
              </a>
            </li>
            <li>
              <a
                class="navbar-li"
                href="/delete_car"
                id=""
              >
                DELETE CAR
              </a>
            </li>
            <li>
              <a
                class="navbar-li"
                href="/reservations"
                id=""
              >
                MY RESERVATIONS
              </a>
            </li>
            <li>
              <a
                class="navbar-li"
                href="/reserve"
                id=""
              >
                RESERVE
              </a>
            </li>
          </ul>
        </div>
        <button
          class="sign-out-btn"
          type="button"
        >
          Log Out
        </button>
        <div
          class="footer"
        >
          <div
            class="NavFooter"
          >
            <footer
              class="text-center text-white"
            >
              <div
                class="container pt-4"
              >
                <section
                  class="mb-4"
                >
                  <a
                    class="btn btn-link btn-floating btn-lg text-dark"
                    data-mdb-ripple-color="dark"
                    href="#!"
                    role="button"
                  >
                    <i
                      class="fa fa-twitter"
                    />
                  </a>
                  <a
                    class="btn btn-link btn-floating btn-lg text-dark"
                    data-mdb-ripple-color="dark"
                    href="#!"
                    role="button"
                  >
                    <i
                      class="fa fa-facebook-f"
                    />
                  </a>
                  <a
                    class="btn btn-link btn-floating btn-lg text-dark"
                    data-mdb-ripple-color="dark"
                    href="#!"
                    role="button"
                  >
                    <i
                      class="fa fa-google-plus"
                    />
                  </a>
                  <a
                    class="btn btn-link btn-floating btn-lg text-dark"
                    data-mdb-ripple-color="dark"
                    href="#!"
                    role="button"
                  >
                    <i
                      class="fa fa-vimeo"
                    />
                  </a>
                  <a
                    class="btn btn-link btn-floating btn-lg text-dark"
                    data-mdb-ripple-color="dark"
                    href="#!"
                    role="button"
                  >
                    <i
                      class="fa fa-pinterest"
                    />
                  </a>
                  <a
                    class="btn btn-link btn-floating btn-lg text-dark"
                    data-mdb-ripple-color="dark"
                    href="#!"
                    role="button"
                  >
                    <i
                      class="fa fa-github"
                    />
                  </a>
                </section>
              </div>
            </footer>
          </div>
        </div>
      </header>
      <main
        class="main"
      >
        <div
          class="session"
        >
          <section
            class="not-allowed"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-triangle-exclamation caution"
              data-icon="triangle-exclamation"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7 .2 40.1S486.3 480 472 480H40c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8 .2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24V296c0 13.3 10.7 24 24 24s24-10.7 24-24V184c0-13.3-10.7-24-24-24zm32 224c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32z"
                fill="currentColor"
              />
            </svg>
            <h2>
              Only Admins Can Add A Car!!!!!
            </h2>
          </section>
        </div>
      </main>
    </div>
    ,
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
